wrapper_writer/scala_parser.py

```diff
@@ -0,0 +1,173 @@
+        :return:
+        """
+        all_found = self.find_method_regex()
+        matches = tuple(all_found)
+        if not matches:
+            raise Exception("No Methods Found")
+        for i in matches:
+            ig = i.group()
+            base_raw = os.path.basename(self.filename)
+            container_name = os.path.splitext(base_raw)[0]
+            return_type = self.extract_return_type(ig)
+            method_name = self.extract_method_name(ig)
+            params = self.extract_params(ig)
+            data = {container_name: {method_name: {"params": params, "returns": return_type}}}
+            with open(self.config_filename, 'a') as yaml_file:
+                yaml.dump(data, yaml_file, default_flow_style=False)
+
+
+class App:
+    """
+    This App class orchestrates the scala parser application
+    :param folder: String object of the folder path which is provided by the user to indicate where files live
+    :param file: String object if the file to be parsed
+    :param file_extension: String object of file extension to look for
+    """
+
+    def __init__(self, folder=None, logic_file=None, file_extension="*.scala", config_name="config_sadhg.yml", append_config=False):
+        self.folder = folder
+        self.file = logic_file
+        self.file_extension = file_extension
+        self.config_file = config_name
+        self.append_config = append_config
+
+    def delete_config(self):
+        """
+        This function will check if a file exist then delete it
+        :return:
+        """
+        print("Checking if file exists ...")
+        if os.path.isfile(self.config_file):
+            os.remove(self.config_file)
+        else:
+            print("The config file does not exists")
+
+    def prepare_input(self):
+        """
+        This function will prepare the use input,
+        :return: all files to be parsed
+        """
+        if not self.append_config:
+            self.delete_config()
+
+        all_files = []
+        prep_ends_with = self.file_extension[1:]
+        if not (self.folder or self.file):
+            raise TypeError("Provide File or Directory")
+        if self.folder and not self.file:
+            if os.path.exists(self.folder):
+                self.folder += self.file_extension
+                for files in glob.glob(self.folder):
+                    if files.endswith(prep_ends_with):
+                        all_files.append(files)
+                return all_files
+            else:
+                print("Throw exception here directory doesnt exist")
+        elif self.file and not self.folder:
+            all_files.append(self.file)
+            return all_files
+
+    def run_scala(self):
+        scala_files_to_run = self.prepare_input()
+        for p in scala_files_to_run:
+            x = ScalaParse(p, self.config_file, self.append_config)
+            x.multi_process()
+
+    def run_python(self):
```

Comment:
When we implement other parser for different languages

Reply:
As we currently dont have them and dont plan to put them in anytime soon, i wouldn't have the functions there as otherwise people might think they are available when they aren't